src/native/libs/System.Native/pal_io.c

```diff
@@ -1472,27 +1472,34 @@ static int16_t ConvertLockType(int16_t managedLockType)
     }
 }
 
-int64_t SystemNative_GetFileSystemType(intptr_t fd)
+uint32_t SystemNative_GetFileSystemType(intptr_t fd)
 {
 #if HAVE_STATFS_VFS || HAVE_STATFS_MOUNT
     int statfsRes;
     struct statfs statfsArgs;
     // for our needs (get file system type) statfs is always enough and there is no need to use statfs64
     // which got deprecated in macOS 10.6, in favor of statfs
     while ((statfsRes = fstatfs(ToFileDescriptor(fd), &statfsArgs)) == -1 && errno == EINTR) ;
-    return statfsRes == -1 ? (int64_t)-1 : (int64_t)statfsArgs.f_type;
+    if (statfsRes == -1) return 0;
+
+    // On Linux, f_type is signed. This causes some filesystem types to be represented as
+    // negative numbers on 32-bit platforms. We cast to uint32_t to make them positive.
+    uint32_t result = (uint32_t)statfsArgs.f_type;
+    assert(result == statfsArgs.f_type);
```

Comment:
@tmds the build was failing with `comparison of integers of different signs: 'uint32_t' (aka 'unsigned int') and '__fsword_t' (aka 'int')`

```suggestion
```

Reply:
This was meant to be a signed against unsigned comparison, like when writing `f_type == SMB2_SUPER_MAGIC`.
The warning could be ignored here because that is what we're trying to check.
If you prefer to leave out the assert, that is also fine for me.